includes/PageVerificationBuilder.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace DataAccounting;
+
+use MediaWiki\MediaWikiServices;
+use MediaWiki\Revision\SlotRecord;
+use MediaWiki\Storage\RevisionRecord;
+
+class PageVerificationBuilder {
+
+	public function buildVerificationData( RevisionRecord $rev ): array {
+		$lb = MediaWikiServices::getInstance()->getDBLoadBalancer();
+		$dbw = $lb->getConnectionRef( DB_MASTER );
+
+		// CONTENT DATA HASH CALCULATOR
+		$pageContent = $rev->getContent( SlotRecord::MAIN )->serialize();
+		$contentHash = getHashSum( $pageContent );
+
+		// GET DATA FOR META DATA and SIGNATURE DATA
+		$parentId = $rev->getParentId();
+		$verificationData = getPageVerificationData( $dbw, $parentId );
+
+		// META DATA HASH CALCULATOR
+		$previousVerificationHash = $verificationData['verification_hash'];
+		$domainId = getDomainId();
+		$timestamp = $rev->getTimeStamp();
+		$metadataHash = calculateMetadataHash( $domainId, $timestamp, $previousVerificationHash );
+
+		// SIGNATURE DATA HASH CALCULATOR
+		$signature = $verificationData['signature'];
+		$publicKey = $verificationData['public_key'];
+		$signatureHash = calculateSignatureHash( $signature, $publicKey );
+
+		// WITNESS DATA HASH CALCULATOR
+		$witnessData = getWitnessData( $verificationData['witness_event_id'] );
+		if ( !empty( $witnessData ) ) {
+			$domain_manifest_verification_hash = $witnessData['witness_event_verification_hash'];
+			$merkle_root = $witnessData['merkle_root'];
+			$witness_network = $witnessData['witness_network'];
+			$witness_tx_hash = $witnessData['witness_event_transaction_hash'];
+			$witnessHash = calculateWitnessHash(
+				$domain_manifest_verification_hash,
+				$merkle_root,
+				$witness_network,
+				$witness_tx_hash
+			);
+		} else {
+			$witnessHash = '';
+		}
+
+		return [
+			'domain_id' => getDomainId(),
+			'page_title' => $rev->getPage()->getDBkey(),
```

Comment:
@JeroenDeDauw is there a reason why you switch to `getDBkey()` from `getTitle()`? Is this because the page is always stored in the form of pagenamee?

Reply:
It is because the WikiPage instance is not available here. Only a RevisionRecord is passed to the method. This is a simple value object, so the code now has a lot less dependencies and is easier to call.

Reply:
OK, I tried `$rev->getPage()->getTitle()` and it caused error with saying that the method `Title::getTitle()` doesn't exist. And then I tried `$rev->getPage()->getPrefixedText()` and it worked, which return prefixed pagename (i.e. with namespace and spaces instead of underscores).